zigpy/application.py

```diff
@@ -191,37 +205,38 @@ def handle_message(
         message: bytes,
         *,
         dst_addressing: Optional[
-            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+            t.Addressing.Group | t.Addressing.IEEE | t.Addressing.NWK
         ] = None,
     ) -> None:
         self.listener_event(
             "handle_message", sender, profile, cluster, src_ep, dst_ep, message
         )
-        if sender.status == zigpy.device.Status.NEW and dst_ep != 0:
-            # only allow ZDO responses while initializing device
-            LOGGER.debug(
-                "Received frame on uninitialized device %s (%s) for endpoint: %s",
-                sender.ieee,
-                sender.status,
-                dst_ep,
-            )
-            zigpy.quirks.handle_message_from_uninitialized_sender(
-                sender, profile, cluster, src_ep, dst_ep, message
-            )
-            return
-        elif (
-            sender.status == zigpy.device.Status.ZDO_INIT
-            and dst_ep != 0
-            and cluster != 0
-        ):
-            # only allow access to basic cluster while initializing endpoints
-            LOGGER.debug(
-                "Received frame on uninitialized device %s endpoint %s for cluster: %s",
-                sender.ieee,
-                dst_ep,
-                cluster,
-            )
-            return
+
+        if not sender.is_initialized:
+            if not sender.initializing:
+                sender.schedule_initialize()
+
+            if not sender.has_node_descriptor and dst_ep != 0:
```

Comment:
Do we need `not sender.has_node_descriptor` here? Doesn't `not sender.is_initialized` satisfies this condition already?

Reply:
Devices that have no node descriptor would be unusable until they're initialized if this check isn't there.

Reply:
Or not, that's what *will* happen with this change.

Maybe it's better to use `if not sender.all_endpoints_init` instead? In that case, the `elif` branch seems like it can be removed, as the presence of initialized endpoints is really the only proxy we have for "probably functional".

I understand the purpose of the first branch, but what is the purpose of the second one? Is it to prevent errors from being logged about missing endpoints and clusters?

Reply:
>  , but what is the purpose of the second one? Is it to prevent errors from being logged about missing endpoints and clusters?

Yes. With the exception of the basic cluster so we could read manufacturer and model. 
But now thinking about it, maybe it is better to allow zdo traffic regardless of the status and ZCL only if all_endpoints_init with the exception of the basic cluster

Reply:
Yeah, that sounds reasonable. I've adjusted `handle_message` accordingly.